shell/platform/windows/window_win32.h

```diff
@@ -217,6 +217,9 @@ class WindowWin32 {
   // Used to process key messages. Exposed for dependency injection.
   virtual uint32_t Win32MapVkToChar(uint32_t virtual_key);
 
+  // Accessor so that it can be mocked in tests.
+  virtual TextInputManagerWin32* get_text_input_manager();
```

Comment:
I wonder if we could do this via injection instead? That would avoid adding a public method here and avoid the possibility of someone accidentally breaking things by using that raw private var in some future change to WindowWin32, and potentially FlutterWindowWin32.

Reply:
OK, I agree that the temptation of using the raw private is possibly bad: I'll redo it as injection. I did think about injection as an option, but the mocking framework wanted to use pointers, and I didn't want to have to inject a heap allocated object, and passing mutable references seems like bad form. I'll probably inject it as a `unique_ptr`.

Reply:
That sounds perfect. Thanks!